File: ace-home-cli/src/app/models/activity.ts
import { Bank } from "../pages/bank/models/bank";

export class Activity {
    id: number;
    description: String;
    amount: number;
    note: String;
    _type: String;
    _date: String;
    bankDto!: Bank;

    constructor(description?: String, amount?: number, note?: String, _type?: String, bankDto?: Bank, _date?: String) {
        this.id = 0;
        description != undefined ? this.description = description : this.description = "";
        amount != undefined ? this.amount = amount : this.amount = 0;
        note != undefined ? this.note = note : this.note = "";
        _type != undefined ? this._type = _type : this._type = "";
        _date != undefined ? this._date = _date : this._date = "";
        bankDto != undefined ? this.bankDto = bankDto : this.bankDto = new Bank;
    }
}

File: ace-home/src/main/java/com/acehome/entities/Activity.java
package com.acehome.entities;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.acehome.entities.bank.Bank;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Entity()
@Table(name="activity")
@Getter
@Setter
@AllArgsConstructor
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column()
    private String description;

    @Column()
    private float amount;

    @Column()
    private String note;

    @Column()
    private String _type;

    @Column()
    private String _date;

    @Column()
    private Boolean _delete;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name="bank_id")
    private Bank  bank;

    public Activity(){
        _delete = false;
    }
    
}
